src/main/kotlin/no/nav/personbruker/dittnav/eventaggregator/common/validation/validationUtil.kt

```diff
@@ -8,4 +10,18 @@ fun timestampToUTCDateOrNull(timestamp: Long?): LocalDateTime? {
     return timestamp?.let { datetime ->
         LocalDateTime.ofInstant(Instant.ofEpochMilli(datetime), ZoneId.of("UTC"))
     }
-}
\ No newline at end of file
+}
+
+fun validatePrefererteKanaler(eksternVarsling: Boolean, field: List<String>): List<String> {
+    val fieldName = "prefererteKanaler"
+    if (!eksternVarsling && field.isNotEmpty()) {
+        throw FieldValidationException("Feltet $fieldName kan ikke settes så lenge eksternVarsling er false.")
+    } else {
+        try {
```

Comment:
Jeg synes denne blir lettere å lese hvis else-branch-en droppes. Slik at det kun er en if for validering, og try-en kommer etterpå.